python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -52,42 +52,82 @@ def destroy_engine(self):
 
     ###########################################################################
     # panel support (displayed as tabs)
-    def _init_tabs(self):
+    def _run_startup_commands(self):
         """
-        Create desktop tabs from the registered application panels, following
-        their specified relative positioning.
-        """
-        # make the "Apps" tab appear as a fake panel to fit in the sorting
-        # logic
-        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
-                    "properties": {"position": 0}}
-
-        panels = [apps_tab] + self._engine.panels.values()
-
-        def panel_position(panel):
-            """ Returns the relative position of a panel """
-            # place the panels with unspecified positions at the end
-            LAST_POSITION = float("inf")
-
-            if "properties" not in panel:
-                return LAST_POSITION
+        Runs the commands that are configured to be executed at startup, through
+        the 'run_at_startup' setting in the configuration.
 
-            return panel["properties"].get("position", LAST_POSITION)
-
-        ordered_panels = sorted(panels, key=panel_position)
-
-        for panel in ordered_panels:
-            # let the panel show itself through its registered callback
-            panel["callback"]()
+        This gives an opportunity for apps to display panels through the
+        execution of a command, which will add a tab to Desktop.
+        """
+        selectors = self._engine.get_setting("run_at_startup", [])
+
+        # "Apps" is currently a builtin command, so we first figure out where it
+        # is (its index) in the selectors:
+        def is_apps_selector(selector):
+            return (selector["app_instance"] == self._engine.instance_name and
+                    (selector["name"] == "" or selector["name"] == "Apps"))
```

Comment:
Is there a reason to include an empty name as a valid name for Apps?  Seems like making it more specific would be better.  Also, maybe have the app_instance be empty for this?  That would be invalid elsewhere so there wouldn't be a conflict and it would make it so that if you renamed the engine instance name for any reason you wouldn't have to go in and update the setting...